datalab/datalab_session/viewsets.py

```diff
@@ -18,6 +19,15 @@ def perform_create(self, serializer):
         serializer.save(session_id=self.kwargs['session_pk'], cache_key=operation.cache_key)
         operation.perform_operation()
 
+    def retrieve(self, request, *args, **kwargs):
+        instance = self.get_object()
+        serializer = self.get_serializer(instance)
+        if instance.status == 'PENDING' and not instance.output:
+            operation = available_operations().get(instance.name)(instance.input_data)
```

Comment:
if an operation is retried but its inputs don't exist yet (because they're also being recreated). Would have to wait or know somehow when the input_data is ready

Reply:
Yeah I guess in our current system it would "Fail" if the inputs didn't exist at that moment and then be removed from the UI so that isn't great. I was trying to avoid making the getting operation logic more complicated to the point where it checks previous operations being done before attempting, but maybe that is the way to go.

Reply:
this kinda ties in to the original plan of having recipes that are saved, thats a whole 'nother can of worms